contracts/programs/gummyroll/src/state/change_log.rs

```diff
@@ -25,78 +28,30 @@ pub struct ChangeLogEvent {
     /// Bitmap of node parity (used when hashing)
     pub index: u32,
 }
-
-#[derive(Copy, Clone, PartialEq)]
-/// Stores proof for a given Merkle root update
-#[repr(C)]
-pub struct ChangeLog<const MAX_DEPTH: usize> {
-    /// Historical root value before Path was applied
-    pub root: Node,
-    /// Nodes of off-chain merkle tree
-    pub path: [Node; MAX_DEPTH],
-    /// Bitmap of node parity (used when hashing)
-    pub index: u32,
-    pub _padding: u32,
-}
-
-impl<const MAX_DEPTH: usize> ChangeLog<MAX_DEPTH> {
-    pub fn to_event(&self, id: Pubkey, seq: u128) -> Box<ChangeLogEvent> {
-        let path_len = self.path.len() as u32;
-        let mut path: Vec<PathNode> = self
+//  ChangeLog<MAX_DEPTH>
+impl<const MAX_DEPTH: usize> From<(Box<ChangeLog<MAX_DEPTH>>, Pubkey, u128)>
+    for Box<ChangeLogEvent>
+{
+    fn from(log_info: (Box<ChangeLog<MAX_DEPTH>>, Pubkey, u128)) -> Self {
+        let (changelog, tree_id, seq) = log_info;
+        let path_len = changelog.path.len() as u32;
+        let mut path: Vec<PathNode> = changelog
             .path
             .iter()
             .enumerate()
-            .map(|(lvl, n)| PathNode::new(*n, (1 << (path_len - lvl as u32)) + (self.index >> lvl)))
+            .map(|(lvl, n)| {
+                PathNode::new(
+                    *n,
+                    (1 << (path_len - lvl as u32)) + (changelog.index >> lvl),
+                )
+            })
```

Comment:
We should add a comment for what the math is doing here and why node_idx is important

Reply:
Addressed in the Gumdrop changes